pkg/providers/apisix/apisix_upstream.go

```diff
@@ -266,55 +266,66 @@ func (c *apisixUpstreamController) sync(ctx context.Context, ev *types.Event) er
 		clusterName := c.Config.APISIX.DefaultClusterName
 		for _, port := range svc.Spec.Ports {
 			for _, subset := range subsets {
-				upsName := apisixv1.ComposeUpstreamName(namespace, name, subset.Name, port.Port)
 				// TODO: multiple cluster
-				ups, err := c.APISIX.Cluster(clusterName).Upstream().Get(ctx, upsName)
-				if err != nil {
-					if err == apisixcache.ErrNotFound {
-						continue
+				update := func(upsName string) error {
```

Comment:
Can it be a function instead of a variable?

Reply:
Yes, but it has many contexts, requires more parameters, and can't be reused. I do not recommend this.